### ISSUE TITLE
Should we scale KL divergence?

### ISSUE THREAD

**Issue by ShellingFord221** [rabeehk/vibert#10]
Hi, in Eq. 2 of your paper, the final loss is the sum of KL divergence and cross-entropy:
![image](https://user-images.githubusercontent.com/40685673/127981526-c33c2232-1f60-4bb4-8498-781f13b804a5.png)

However, in implement, we feed the network with a batch of data each time, i.e. 
https://github.com/rabeehk/vibert/blob/c2e7aa3f81ff9a73eb112739e284cb98b6fa2de4/run_glue.py#L198

While KL divergence is calculated across the whole data distribution:
![image](https://user-images.githubusercontent.com/40685673/127982334-3c58fc50-2d85-4c15-998c-10a0d40dd638.png)

This will cause that KL divergence is not in the same scale of cross-entropy. However, it is easy to scale KL divergence according to the batch_size of training data:
![image](https://user-images.githubusercontent.com/40685673/127982537-90faa837-67eb-40c4-9bf6-18014905bd90.png)
 (i.e. Eq. 8 of [Weight uncertainty in Neural Networks](https://arxiv.org/pdf/1505.05424.pdf))

Then we can multiply scaled KL divergence with parameter \beta as the final loss of information bottleneck for line 179 in models.py:
> loss["loss"] = ce_loss + **1 / M** * (beta if self.kl_annealing == "linear" else self.beta) * kl_loss

I know that scaled term `1/M` can be considered as being integrated into \beta, but will this cause strange value of \beta (e.g. a quite small \beta) and unstable training process? Thanks!

**Comment by rabeehk** [comment]
Hi @ShellingFord221 
Thank you for your comment. We have at the end kl_divergence.mean() (to average over the batch as you mentioned), do you think it might address your comment? 
thanks.
Best
Rabeeh
